src/d3a/models/strategy/__init__.py

```diff
@@ -104,6 +104,23 @@ def is_offer_posted(self, market_id, offer_id):
         return offer_id in [offer.id for offer, _market in self.posted.items()
                             if market_id == _market]
 
+    def get_sold_offer_ids_in_market(self, market_id):
+        sold_offer_ids = []
+        for sold_offer in self.sold.get(market_id, []):
+            sold_offer_ids.append(sold_offer.id)
+        return sold_offer_ids
+
+    def open_in_market(self, market_id):
+        open_offer = []
+        sold_offer_ids = self.get_sold_offer_ids_in_market(market_id)
+        for offer, _market in self.posted.items():
```

Comment:
As far as I understand, the reason for introducing these methods is in order to be able to calculate the energy of the existing offers. This is exactly what the posted offers dict keeps, essentially when an offer is deleted or traded, it should be removed from the posted dict. That's why I do not think that you need any of these 3 functions (`get_sold_offer_ids_in_market `, `open_in_market `, `open_offer_energy `), you would achieve the same result by using the `posted_offer_energy` method.
cc'ing @hannesdiedrich who has recently worked in this class.

Reply:
Yes, I agree, `posted_offer_energy` should be sufficient.

Reply:
`posted_in_market` reports all the offer that has been yet posted (it is kept in `posted` even after the trade OR being deleted) whereas for `offered_sell_kWh` of ESS needs the offered energy that are not traded yet.

Reply:
Judging from the doc-string of this [class](https://github.com/gridsingularity/d3a/pull/685/files#diff-5c5fbb99e0155bbce18f8ca72347fd58R59), it is expected to distinguish `open_offer`  from the `posted_offer`

Reply:
Unfortunately the link does not work in your case, but I think I found the culprit here:
https://github.com/gridsingularity/d3a/blob/master/src/d3a/models/strategy/__init__.py#L150
Therefore indeed the posted offers do contain the sold offers. In this case this works for me, and these functions serve a purpose. Thanks!

Reply:
Hmm, ok, so self.posted keeps track of ALL offers, including the sold and active ones. 
Why do we have this again? This is probably heritage from before, right? Anyways. 
I agree with Spyros. Leave your implementation. Thanks!

Reply:
There is indeed a room for potential refactoring as I see in case of `bids` inside `BidEnabledStrategy`, there being popped-out from `_bids` dict after being traded.
If you guys agree I would be very much in favour of a future enhancement story

Reply:
I agree that the behavior of both should be synced, feel free to create the enhancement story. Thanks!